src/Plugin/Field/FieldFormatter/StrawberryPagedFormatter.php

```diff
@@ -562,14 +496,23 @@ public function processElementforManifestURL(
           $element['media']['#attached']['drupalSettings']['format_strawberryfield']['iabookreader'][$htmlid]['nodeuuid'] = $nodeuuid;
           $element['media']['#attached']['drupalSettings']['format_strawberryfield']['iabookreader'][$htmlid]['manifesturl'] = $manifest_url;
           $element['media']['#attached']['drupalSettings']['format_strawberryfield']['iabookreader'][$htmlid]['width'] = $max_width_css;
-          $element['media']['#attached']['drupalSettings']['format_strawberryfield']['iabookreader'][$htmlid]['height'] = max(
-            $max_height,
-            520
-          );
+          $element['media']['#attached']['drupalSettings']['format_strawberryfield']['iabookreader'][$htmlid]['height'] = max($max_height, 520);
+          // @see self::processElementforMetadatadisplays()
+          $element['media']['#attached']['drupalSettings']['format_strawberryfield']['iabookreader'][$htmlid]['enableSearchArchipelago'] = $this->searchEnabled($item);
```

Comment:
To spare us from some duplicated code, what about (looking at this now for the first time with 2021 eyes) we make a check on the caller method (Like line 334, inside the $delta loop) and check 
```PHP
if (isset($element[$delta]['media'])) {
 // all the common settings?
}
```
what do you think? I mean since you have done so much good work cleaning this old code, maybe that would save is a few lines.

Reply:
I used to do something like `$js_settings = &$element['media']['#attached']['drupalSettings']['format_strawberryfield']['iabookreader'][$htmlid]` but I've found is not the most stable or secure pattern as it can get easily polluted.

What I'll do is to transform this in a single array declaration which looks nicer, but we might want to consider having a base method with sensible defaults that is then called by the manifest and manifest url methods as there's a fair amount of code shared between both paths. I'd do in a followup, though